docs/arch/adr-37.md

```diff
@@ -0,0 +1,30 @@
+# ADR 37: Engaging Crowds: Indexed subject sets
+
+7 December 2021
+
+## Context
+
+Engaging Crowds allows volunteers to browse and search subject sets, linked to a workflow, in order to pick which subjects they want to work on. We needed a solution that would allow us to search a set, via indexed subject metadata, and present the results to a volunteer.
+
+The subjects table in Panoptes is too large to allow for fast querying, so we also needed to build our own indexing system for indexed sets.
+
+## Decision
+
+- Project owners can flag subject metadata columns as searchable by prefixing the heading with `%` in the manifest eg. these manifest headings `subject_id,image_name_1,%origin,link,%attribution,license,#secret_description` mark `metadata.origin` and `metadata.attribution` as searchable.
+- Subject sets with indexed subjects have `metadata.indexFields` set to a list of indexed fields eg. `indexFields: 'origin,attribution'`.
+- Subject metadata for indexed sets is copied to a separate database running on [Datasette](https://datasette.io). Each set is given its own table, named by subject set ID. Datasette gives us a RESTful API out of the box, allowing us to browse and search subject data as HTML or JSON. See https://subject-set-search-api.zooniverse.org/subjects.
+- From a volunteers point-of-view, the Datasette service is used to find specific subject IDs to work on. Those IDs are then sent to the Panoptes API `/subjects/selection` endpoint, which returns those subjects, in order, for classification.
+
+## Consequences
```

Comment:
Suggestion: add to "Consequences" section information about lack of pagination for the Indexing pages, leading us to require upper limits for subject sets per workflow (50) and subjects per set (100).

Reply:
Good point. The workflows menu can only show one page (20 workflows) too.

Reply:
Volunteers can use the subject search to explore sets larger than 100 subjects, but I'm not sure how obvious or useful that is eg. enter '1884' in a Date column to see 100 subjects from 1884. That sort of thing.